client.go

```diff
@@ -682,6 +682,10 @@ func (c *Client[TTx]) signalStopComplete(ctx context.Context) {
 // by cancelling the context passed to Start or by calling StopAndCancel.
 func (c *Client[TTx]) Stop(ctx context.Context) error {
 	c.baseService.Logger.InfoContext(ctx, c.baseService.Name+": Stop started")
+	if c.fetchNewWorkCancel != nil {
+		return errors.New("client not started")
+	}
```

Comment:
@pmenglund What do you think about moving this to be above the log info line? I think it makes more sense for the log line to only be written in cases where a stop really is starting (i.e. not a false start).